Sources/PriorityQueueModule/PriorityQueue+Invariants.swift

```diff
@@ -0,0 +1,91 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Collections open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+import Foundation
+
+extension PriorityQueue {
+  #if COLLECTIONS_INTERNAL_CHECKS
+  @inlinable
+  @inline(never)
+  // Iterates through all the levels in the heap, ensuring that items in min
+  // levels are smaller than all their descendants and items in max levels
+  // are larger than all their descendants.
+  //
+  // The min-max heap indices are structured like this:
+  //  min               0
+  //  max        1             2
+  //  min      3   4         5     6
+  //  max     7 8 9 10     11 12 13 14
+  //  min    15...
+  //  ...
+  //
+  // The iteration happens in depth-first order, so the descendants of the
+  // element at index 0 are checked to ensure they are >= that element. This is
+  // repeated for each child of the element at index 0 (inverting the
+  // comparison at each level).
+  //
+  // In the case of 7 elements total (spread across 3 levels), the checking
+  // happens in the following order:
+  //   compare >= 0: 1, 3, 4, 2, 5, 6
+  //   compare <= 1: 3, 4
+  //   compare >= 3: (no children)
+  //   compare >= 4: (no children)
+  //   compare <= 2: 5, 6
+  //   compare >= 5: (no children)
+  //   compare >= 6: (no children)
+  internal func _checkInvariants() {
```

Comment:
> Since heaps have some invariants that need to be maintained (min-max heaps especially), you could consider adding a method that checks these invariants just for testing purposes. With some of the operations it can be a bit tricky to keep a clear mental model of exactly what's happening.

@timvermeulen Is this what you had in mind?

Reply:
Absolutely, this looks great. I was imagining we call this method from the tests somewhere but since the mutating operations already call it themselves, I don't think we need to.